manager/controlapi/service.go

```diff
@@ -170,6 +170,22 @@ func validateEndpointSpec(epSpec *api.EndpointSpec) error {
 	return nil
 }
 
+func (s *Server) validateNetworks(networks []*api.NetworkAttachmentConfig) error {
+	for _, na := range networks {
+		var network *api.Network
+		s.store.View(func(tx store.ReadTx) {
+			network = store.FindNetwork(tx, na.Target)
+		})
+		if network == nil {
+			continue
+		}
+		if _, ok := network.Spec.Annotations.Labels["com.docker.swarm.internal"]; ok {
+			return grpc.Errorf(codes.InvalidArgument, "Service cannot be explicitly connected to the ingress network")
```

Comment:
The error message  probably should extract the name from the network since may be in the future we may have more than one internal network. Also It should probably be rephrased to something like `Service cannot be explicitly attached to "ingress" network which is an internal network`